x-pack/legacy/plugins/uptime/public/state/api/monitor_duration.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { stringify } from 'query-string';
+
+import { getApiPath } from '../../lib/helper';
+import { BaseParams } from './types';
+
+export const fetchMonitorDuration = async ({
+  basePath,
+  monitorId,
+  dateStart,
+  dateEnd,
+}: BaseParams) => {
+  const url = getApiPath(`/api/uptime/monitor/duration`, basePath);
+
+  const params = {
+    monitorId,
+    dateStart,
+    dateEnd,
+  };
+  const urlParams = stringify(params);
+
+  const response = await fetch(`${url}?${urlParams}`);
```

Comment:
I want to make a generic implementation of this logic as it's repeated exactly in many API functions. WDYT? Not part of this PR obviously.

Reply:
I created an issue to track this https://github.com/elastic/uptime/issues/153.

Reply:
agreed, we can abstract this.